template-coq/src/g_template_coq.mlg

```diff
@@ -196,11 +195,16 @@ TACTIC EXTEND TemplateCoq_run_template_program
          let env = Proofview.Goal.env gl in
          let evm = Proofview.Goal.sigma gl in
          let ret = ref None in
-         Run_template_monad.run_template_program_rec ~poly ~intactic:true (fun x -> ret := Some x) env (evm, to_constr_evars evm c);
+         (* We don't allow opening obligations / updating the vernacular inside proofs / as tactics *)
+         let pm = Declare.OblState.empty in
+         let _pm = Run_template_monad.run_template_program_rec
```

Comment:
Maybe should assert the returned pm is empty?

Reply:
Indeed we can tweak this a bit, also on the lines suggested above by Matthieu / Yannick; I may give it a go.